app/bundles/CampaignBundle/Entity/LeadEventLogRepository.php

```diff
@@ -677,4 +678,49 @@ public function deleteAnonymousContacts(): int
 
         return $deletedRecordCount;
     }
+
+    public function getEventLogStats(int $eventId): EventLogStatsDto
+    {
+        $qb = $this->getReplicaConnection()->createQueryBuilder();
+        $qb->select([
+            'COUNT(log.id) as total_logs',
+            'COUNT(DISTINCT log.lead_id) as unique_executions',
+            'SUM(log.is_scheduled) as pending_executions',
+            'SUM(CASE WHEN log.non_action_path_taken = 1 AND log.is_scheduled = 0 THEN 1 ELSE 0 END) as negative_path_count',
+            'SUM(CASE WHEN log.non_action_path_taken = 0 AND log.is_scheduled = 0 THEN 1 ELSE 0 END) as positive_path_count',
+            'MIN(log.date_triggered) as first_execution_date',
+            'MAX(log.date_triggered) as last_execution_date',
+            'MAX(log.rotation) as max_rotations',
+        ])
+            ->from(MAUTIC_TABLE_PREFIX.'campaign_lead_event_log', 'log')
+            ->where(
+                $qb->expr()->and(
+                    $qb->expr()->eq('log.event_id', ':eventId')
+                )
+            )
+            ->setParameter('eventId', $eventId)
+            ->groupBy('log.event_id');
+
+        $result = $qb->executeQuery()->fetchAssociative();
+
+        if (false === $result) {
+            return new EventLogStatsDto();
+        }
+
+        $totalLogs         = (int) ($result['total_logs'] ?? 0);
+        $pendingExecutions = (int) ($result['pending_executions'] ?? 0);
+        $uniqueExecutions  = (int) ($result['unique_executions'] ?? 0);
+        $maxRotations      = (int) ($result['max_rotations'] ?? 0);
+
+        return new EventLogStatsDto(
+            totalExecutions: $totalLogs - $pendingExecutions,
+            uniqueExecutions: $uniqueExecutions,
+            pendingExecutions: $pendingExecutions,
+            maxRotations: $maxRotations,
+            negativePathCount: (int) ($result['negative_path_count'] ?? 0),
+            positivePathCount: (int) ($result['positive_path_count'] ?? 0),
+            firstExecutionDate: $result['first_execution_date'] ? new \DateTime($result['first_execution_date']) : null,
+            lastExecutionDate: $result['last_execution_date'] ? new \DateTime($result['last_execution_date']) : null
```

Comment:
It's recommended to use immutable date objects. Unless there is a good reason to have mutable ones which sadly happens in Mautic due the legacy code.
```suggestion
            firstExecutionDate: $result['first_execution_date'] ? new \DateTimeImmutable($result['first_execution_date']) : null,
            lastExecutionDate: $result['last_execution_date'] ? new \DateTimeImmutable($result['last_execution_date']) : null
```